diff --git a/utils/aplay/aplay.c b/utils/aplay/aplay.c
--- a/utils/aplay/aplay.c
+++ b/utils/aplay/aplay.c
@@ -527,5 +527,6 @@
 	struct pollfd pfds[] = {{ w->ba_pcm_fd, POLLIN, 0 }};
 	int timeout = -1;
+	bool fifo_closed = false;
 
 	debug("Starting PCM loop");
 	while (main_loop_on) {
@@ -577,9 +578,8 @@
 		}
 
 		if (input_ready) {
-			/* FIFO has been terminated on the writing side */
 			if (pfds[0].revents & POLLHUP)
-				break;
+				fifo_closed = true;
 
 			#define MIN(a,b) a < b ? a : b
 			size_t _in = MIN(pcm_max_read_len, ffb_blen_in(&buffer));
@@ -592,6 +592,9 @@
 			}
 		}
 
+		if (fifo_closed && ret == 0 && ffb_len_out(&buffer) == 0)
+			break;
+
 		pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, NULL);
 
 		/* If PCM mixer is disabled, check whether we should play audio. */
@@ -676,7 +679,8 @@
 		snd_pcm_sframes_t frames = ffb_len_out(&buffer) / w->ba_pcm.channels;
 
 		if (!pcm_started) {
-			if (frames < (snd_pcm_sframes_t)pcm_jitter_frames) {
+			if (!fifo_closed &&
+					frames < (snd_pcm_sframes_t)pcm_jitter_frames) {
 				timeout = pcm_prefill_timeout;
 				continue;
 			}
@@ -684,7 +688,10 @@
 		}
 		timeout = pcm_inactive_timeout;
 
-		if (frames < (snd_pcm_sframes_t)pcm_period_frames)
+		if (frames < (snd_pcm_sframes_t)pcm_period_frames) {
+			if (fifo_closed)
+				break;
 			continue;
+		}
 		frames = pcm_period_frames;
 
