--- a/drivers/spi/spi-dw.c
+++ b/drivers/spi/spi-dw.c
@@ -417,7 +417,12 @@ static void pump_transfers(unsigned long data)
 			udelay(previous->delay_usecs);
 	}
 	gpioreg = (volatile u32*)0xF7E80450;
-               while((*gpioreg & 0x2000) == 0);
+	if ((*gpioreg & 0x2000) == 0) {
+		dev_err_ratelimited(&dws->master->dev,
+			"GPIO ready was not asserted\n");
+		message->status = -ETIMEDOUT;
+		goto early_exit;
+	}
 
 	dws->n_bytes = chip->n_bytes;
 	dws->dma_width = chip->dma_width;
